src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs

```diff
@@ -30,6 +31,18 @@ internal partial class SqlConnection
         private readonly sqlite3 _handle;
 
         /// <summary>
+        /// Whether or not we were able to successfully instantiate this connection. If we weren't
+        /// then we definitely do not want to report any issues about the connection not being
+        /// closed up as the caller won't have an actual instance to close.
+        /// </summary>
+        private readonly bool _successfullyCreated;
```

Comment:
That was one of the options i mentioned.  I didn't think it was much bette rthan this approach.  But i can change it if you feel you'd prefer it.